tests/santa/test_api_views.py

```diff
@@ -713,6 +723,632 @@ def test_rule_list_by_unknown_configuration(self):
+        rule.refresh_from_db()
+        self.assertNotIn(tag_conflicts, rule.tags.all())
+        self.assertNotIn(tag_conflicts, rule.excluded_tags.all())
+
+    def test_update_rule_target_does_not_exist_error(self):
+        configuration = self.force_configuration()
+        target_identifier = get_random_string(length=64, allowed_chars='abcdef0123456789')
+        rule = self.force_rule(configuration=configuration, target_identifier=target_identifier)
+        self.set_permissions("santa.change_rule")
+        data = {
+            "configuration": self.configuration.pk,
+            "policy": Rule.ALLOWLIST,
+            "target_type": Target.BINARY,
+            "target_identifier": get_random_string(length=64, allowed_chars='abcdef0123456789')
+        }
+        response = self.put_json_data(reverse("santa_api:rule", args=(rule.pk,)), data)
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(response.json(), {'target': ['does not exist!']})
+        rule.refresh_from_db()
+        self.assertEqual(rule.target.identifier, target_identifier)
+
+    def test_update_rule_change_custom_msg(self):
+        configuration = self.force_configuration()
+        rule = self.force_rule(configuration=configuration, target_type=Target.BINARY, policy=Rule.BLOCKLIST)
+        self.set_permissions("santa.change_rule")
+        data = {
+            "configuration": self.configuration.pk,
+            "policy": rule.policy,
+            "target_type": rule.target.type,
+            "target_identifier": rule.target.identifier,
+            "description": rule.description,
+            "custom_msg": "new custom message"
+        }
+        response = self.put_json_data(reverse("santa_api:rule", args=(rule.pk,)), data)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        rule.refresh_from_db()
+        self.assertEqual(rule.custom_msg, "new custom message")
+        self.assertEqual(rule.version, 2)
+
+    def test_update_rule_not_found(self):
+        self.set_permissions("santa.change_rule")
+        response = self.put_json_data(reverse("santa_api:rule", args=(1234567890,)), {})
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_update_rule_permission_denied(self):
+        configuration = self.force_configuration()
+        rule = self.force_rule(configuration=configuration)
+        response = self.put_json_data(reverse("santa_api:rule", args=(rule.pk,)), {})
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_update_rule_unauthorized(self):
+        configuration = self.force_configuration()
+        rule = self.force_rule(configuration=configuration)
+        response = self.put_json_data(reverse("santa_api:rule", args=(rule.pk,)), {}, include_token=False)
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
```

Comment:
You should add tests to trigger all the conditions in the building of the update rule event.